File: src/plugin/command/server_assistant/KickCommandPlugin.java
package plugin.command.server_assistant;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.plugin.type.CommandPlugin;
import org.redrune.game.global.World;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Kicks a player from the server", types = { String.class })
public class KickCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		String name = getCompleted(args, 1);
		Player target = World.getPlayerByDisplayName(name);
		if (target == null) {
			return;
		}
		target.getSession().getChannel().close();
		World.removePlayer(target);
	}
	
	@Override
	public String[] identifiers() {
		return arguments("kick");
	}
}

File: src/main/kotlin/plugin/command/server_assistant/KickCommandPlugin.kt
package plugin.command.server_assistant

import plugin.command.CommandManifest
import org.redrune.game.content.plugin.type.CommandPlugin
import org.redrune.game.entity.actor.player.Player
import org.redrune.game.global.World

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Kicks a player from the server", types = [String::class])
class KickCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val name = getCompleted(args, 1)
        val target = World.getPlayerByDisplayName(name) ?: return
        target.session.channel.close()
        World.removePlayer(target, false)
    }

    override fun identifiers(): Array<String> {
        return arguments("kick")
    }
}